core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -381,11 +378,7 @@ public final boolean isEmpty() {
    }
 
    final boolean isEmpty(EnumSet<Flag> explicitFlags, ClassLoader explicitClassLoader) {
-      // TODO: replace with stream findAny isPresent instead
-      try (CloseableIterable<CacheEntry<K, Void>> iterable = filterEntries(AcceptAllKeyValueFilter.getInstance(),
-            explicitFlags, explicitClassLoader).converter(NullValueConverter.getInstance())) {
-         return !iterable.iterator().hasNext();
-      }
+      return !entrySet(explicitFlags, explicitClassLoader).stream().anyMatch(StreamMarshalling.alwaysTruePredicate());
```

Comment:
Unfortunately doing so would require putting the implementation in 4 different places.  I am hoping to redo this at some point as it would be more efficient then having to go back into the cache to again.  I was thinking I would refactor that when I changed the iterator/spliterator stuff as I have to rewrite a good chunk of related code anyways.